### ISSUE TITLE
Suggestion: TypeScript Support

### ISSUE THREAD

**Issue by ryanewtaylor** [brygrill/wab-react-webpack#6]
Firstly,  this repository is fantastic. Thanks sharing this! I discovered this a few weeks ago and brought it to our team. We're now trying it out on a few small projects as we believe it'll really improve our development workflow. We already use WAB, React, and Semantic UI React so seems like a perfect fit. 

However, we develop with TypeScript when afforded the opportunity. I have experimented adding TypeScript support to the v1.1.1 release (we cannot yet support hooks) and have had some success but it will need more review from my team and others.

Do you have plans (or interest) in supporting TypeScript development within this repository? Would you be open to a PR adding TypeScript support, or would a fork be more appropriate?

**Comment by brygrill** [comment]
That is awesome to hear! Would love a PR for TypeScript. That would be really good, as I know it is widely used in the Esri ecosystem. Is it just a matter of adding additional loaders to webpack?

**Comment by ryanewtaylor** [comment]
I'm still working through some challenges with a test conversion but Babel 7 supports TypeScript which makes the conversion a little easier. 

I have followed (most of) the steps defined in https://iamturns.com/typescript-babel/ and https://blogs.msdn.microsoft.com/typescript/2018/08/27/typescript-and-babel-7/. 

Beyond that there were some additional tweaks I had to make so that type checks would pass. Importing new type definitions, some modifications to exports, and so on. The changes required seem to depend on how strict one wants to be.

FWIW, I have only attempted converting *.js in the src/components folder + the index.js file. It wouldn't make sense to convert the source WAB app files.

A PR might help illuminate the changes and help discussion. I'd be glad to help if you are still interested.
